File: src/php/bubbleSort.php
<?php

/**
 * 冒泡排序
 * @param array $array
 */
function bubbleSort($array)
{
    $count = count($array);
    for ($i = 0; $i < $count; $i ++) {
        for ($j = $i + 1; $j < $count; $j ++) {
            if ($array[$i] > $array[$j]) {
                $temp = $array[$i];
                $array[$i] = $array[$j];
                $array[$j] = $temp;
            }
        }
    }
    return $array;
}

$arr = array(
    49,
    38,
    65,
    97,
    76,
    13,
    27,
    50
);

$output = bubbleSort($arr);
print_r($output);

File: src/go/bubbleSort.go
package main

import (
	"fmt"
)

// 冒泡排序
func bubbleSort(arr []int) {
	times := len(arr)
	for i := 1; i < len(arr); i++ {
		for i := 1; i < times; i++ {
			if arr[i-1] > arr[i] {
				temp := 0
				temp = arr[i-1]
				arr[i-1] = arr[i]
				arr[i] = temp
			}
		}
		times--
	}
}

func main() {
	// 设置一个切片进行测试
	arr1 := []int{24, 69, 80, 57, 13}
	bubbleSort(arr1)
	fmt.Println(arr1)

	// 给切片添加新的数字，进一步测试
	arr2 := append(arr1, 45, 97, 86)
	bubbleSort(arr2)
	fmt.Println(arr2)
}
